packages/ember-metal-views/lib/renderer.js

```diff
@@ -210,7 +210,7 @@ Renderer.prototype.renderElementRemoval =
     if (view._willRemoveElement) {
       view._willRemoveElement = false;
 
-      if (view._renderNode) {
+      if (view._renderNode && view.element && view.element.parentNode) {
```

Comment:
This is the especially hacky bit. Without this, views that are already out-of-DOM will delete unrelated subsequent content that has replaced them in the same render node.